packages/flutter/test/widgets/selectable_region_test.dart

```diff
@@ -3321,14 +3332,73 @@ void main() {
         // Test doesn't run these platforms.
         break;
     }
+  },
+    variant: TargetPlatformVariant.mobile(),
+    skip: kIsWeb, // [intended] Web uses its native context menu.
+  );
 
+  testWidgets('Selection behavior when clicking the `Share` button on Android', (WidgetTester tester) async {
+    List<ContextMenuButtonItem> buttonItems = <ContextMenuButtonItem>[];
+    final FocusNode focusNode = FocusNode();
+    addTearDown(focusNode.dispose);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: SelectableRegion(
+          focusNode: focusNode,
+          selectionControls: materialTextSelectionHandleControls,
+          contextMenuBuilder: (
+            BuildContext context,
+            SelectableRegionState selectableRegionState,
+          ) {
+            buttonItems = selectableRegionState.contextMenuButtonItems;
+            return const SizedBox.shrink();
+          },
+          child: const Text('How are you?'),
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+
+    final RenderParagraph paragraph = tester.renderObject<RenderParagraph>(
+      find.descendant(
+        of: find.text('How are you?'),
+        matching: find.byType(RichText),
+      ),
+    );
+    await tester.longPressAt(textOffsetToPosition(paragraph, 6)); // at the 'r'
+    await tester.pump(kLongPressTimeout);
+
+    // `are` is selected.
+    expect(paragraph.selections[0], const TextSelection(baseOffset: 4, extentOffset: 7));
+
+    String? lastShare;
+    TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
+        .setMockMethodCallHandler(SystemChannels.platform, (MethodCall methodCall) async {
+      if (methodCall.method == 'Share.invoke') {
+        expect(methodCall.arguments, isA<String>());
+        lastShare = methodCall.arguments as String;
+      }
+      return null;
+    });
+    addTearDown(() => tester.binding.defaultBinaryMessenger.setMockMethodCallHandler(SystemChannels.platform, null));
+
+    final SelectableRegionState regionState = tester.state<SelectableRegionState>(find.byType(SelectableRegion));
+
+    // Press the `Share` button.
+    expect(buttonItems[1].type, ContextMenuButtonType.share);
+    buttonItems[1].onPressed?.call();
+    expect(lastShare, 'are');
+    // On Android, share should clear the selection.
+    expect(regionState.selectionOverlay, isNull);
+    expect(regionState.selectionOverlay?.startHandleLayerLink, isNull);
+    expect(regionState.selectionOverlay?.endHandleLayerLink, isNull);
   },
-    skip: kIsWeb, // [intended]
-    variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS, TargetPlatform.android, TargetPlatform.fuchsia }),
+    skip: kIsWeb, // [intended] Web uses its native context menu.
```

Comment:
should we only run this test on `Android`?

Reply:
Yes this test is Android only on purpose. This is the new test I added.
The Github diff makes it look like I removed the variant line for it, but this is a variant line I updated for another test (see lines 3336-3337).

Testing if the share button is visible on other platforms is done in a test below (lines 3440-3455).
This new test checks the behavior when clicking on the share button (this is why it is Android only).